clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java

```diff
@@ -900,6 +901,37 @@ public void testOffsetCommitAsyncFailedWithRetriableThrowsRetriableCommitExcepti
         assertFutureThrows(commitResult, RetriableCommitFailedException.class);
     }
 
+    @ParameterizedTest
+    @MethodSource("offsetCommitExceptionSupplier")
+    public void testOffsetCommitOnFailAttemptsWhenPartitionError(final Errors error) {
+        CommitRequestManager commitRequestManager = create(true, 100);
+        when(coordinatorRequestManager.coordinator()).thenReturn(Optional.of(mockedNode));
+
+        Map<TopicPartition, OffsetAndMetadata> offsets = new HashMap<>();
+        offsets.put(new TopicPartition("t1", 1), new OffsetAndMetadata(1));
+        offsets.put(new TopicPartition("t1", 2), new OffsetAndMetadata(2));
+        offsets.put(new TopicPartition("t1", 3), new OffsetAndMetadata(3));
+
+        commitRequestManager.commitSync(offsets, time.milliseconds() + defaultApiTimeoutMs);
+        NetworkClientDelegate.PollResult res = commitRequestManager.poll(time.milliseconds());
+        assertEquals(1, res.unsentRequests.size());
+        List<OffsetCommitResponseData.OffsetCommitResponsePartition> partitions = new ArrayList<>();
+        partitions.add(new OffsetCommitResponseData.OffsetCommitResponsePartition().setErrorCode(error.code()).setPartitionIndex(1));
+        partitions.add(new OffsetCommitResponseData.OffsetCommitResponsePartition().setErrorCode(error.code()).setPartitionIndex(2));
+        partitions.add(new OffsetCommitResponseData.OffsetCommitResponsePartition().setErrorCode(error.code()).setPartitionIndex(3));
+
+        res.unsentRequests.get(0).handler().onComplete(mockOffsetCommitResponse("topic", partitions, (short) 1));
+        CommitRequestManager.OffsetCommitRequestState commitRequest = commitRequestManager.pendingRequests.unsentOffsetCommits.peek();
+        if (error.exception() instanceof RetriableException) {
+            assertNotNull(commitRequest);
+            assertEquals(1, commitRequest.numAttempts, "Request should only retry once no matter how many partition error");
```

Comment:
I guess the message here is more about failed attempts registered (not retries). Something like: only one failed attempt should be registered for the request even if the response contains many partition errors